src/hooks/useOnClickOutside.ts

```diff
@@ -1,18 +1,24 @@
 import { useEffect } from "react";
 
 export const useOnClickOutside = (
-  ref: React.RefObject<HTMLElement>,
-  cb: () => void
+  refs: Array<React.RefObject<HTMLElement>>,
+  cb: (isFocused: boolean) => void,
+  isOpen: boolean
 ): void => {
   useEffect(() => {
     function handleClickOutside(event): void {
-      if (ref.current && !ref.current.contains(event.target)) {
-        cb();
+      if (!isOpen) {
+        return;
       }
+      // Array.some returns true if some value within array satisfies condition, false otherwise
+      const isFocused = refs.some((ref) =>
+        ref.current!.contains(event.target as Node)
+      );
+      cb(isFocused);
     }
     document.addEventListener("mousedown", handleClickOutside);
     return (): void => {
       document.removeEventListener("mousedown", handleClickOutside);
     };
-  }, [cb, ref]);
+  }, [cb, refs, isOpen]);
 };
```

Comment:
I pretty much just moved the existing code into the hook... I tested with one ref and two refs, and it works as I would expect